dali/c_api/c_api_test.cc

```diff
@@ -622,9 +622,9 @@ TYPED_TEST(CApiTest, TestExecutorMeta) {
 
   pipe_ptr.reset();
   daliPipelineHandle handle;
-  daliCreatePipeline(&handle, serialized.c_str(), serialized.size(), batch_size, num_thread,
-                     this->device_id_, false, prefetch_queue_depth, prefetch_queue_depth,
-                     prefetch_queue_depth, true);
+  daliCreatePipeline2(&handle, serialized.c_str(), serialized.size(), batch_size, num_thread,
+                      this->device_id_, false, false, false,
+                      prefetch_queue_depth, prefetch_queue_depth, prefetch_queue_depth, true);
```

Comment:
Executor 2.0 doesn't provide ExecutorMeta and doesn't support buffer preallocation (because it's against it's very design) - and it can forcibly replace the async-pipelined one if the environment variable is specified. Hence, use non-async, non-pipelined executor in meta and preallocation tests.